smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/StructuredMemberWriter.java

```diff
@@ -65,6 +70,173 @@ void writeMembers(TypeScriptWriter writer, Shape shape) {
+                writer.write("...$L,", accParam);
+                Shape memberTarget = model.expectShape(mapMember.getTarget());
+                writer.openBlock("[$L]: ", ",", keyParam, () -> {
+                    if (memberTarget instanceof StructureShape) {
+                        writeStructureFilterSensitiveLog(writer, memberTarget, valueParam);
+                    } else if (memberTarget instanceof CollectionShape) {
+                        MemberShape collectionMember = ((CollectionShape) memberTarget).getMember();
+                        writeCollectionFilterSensitiveLog(writer, collectionMember, valueParam);
+                    } else if (memberTarget instanceof MapShape) {
+                        MemberShape nestedMapMember = ((MapShape) memberTarget).getValue();
+                        writeMapFilterSensitiveLog(writer, nestedMapMember, valueParam);
+                    } else {
+                        // This path should not reach because of recursive isIterationRequired.
+                        throw new CodegenException(String.format(
+                            "MapFilterSensitiveLog attempted for %s while it was not required",
+                            memberTarget.getType()
+                        ));
+                        // For quick-fix in case of high severity issue:
+                        // comment out the exception above and uncomment the line below.
+                        // writer.write("$L", valueParam);
+                    }
+
+                });
+            }
+        );
+    }
+
+    /**
+     * Identifies if iteration is required on member.
+     *
+     * @param member a {@link MemberShape} to check for iteration required.
+     * @return Returns true if the iteration is required on member.
+     */
+    private boolean isIterationRequired(MemberShape member) {
+        Shape memberTarget = model.expectShape(member.getTarget());
+        if (memberTarget instanceof StructureShape) {
+            return true;
+        } else if (memberTarget instanceof CollectionShape) {
+            MemberShape collectionMember = ((CollectionShape) memberTarget).getMember();
+            return isIterationRequired(collectionMember);
+        } else if (memberTarget instanceof MapShape) {
+            MemberShape mapMember = ((MapShape) memberTarget).getValue();
+            return isIterationRequired(mapMember);
+        }
+        return false;
+    }
+
+    /**
+     * Identifies if member needs to be overwritten in filterSensitiveLog.
+     *
+     * @param member a {@link MemberShape} to check if overwrite is required.
+     * @return Returns true if the overwrite is required on member.
+     */
+    private boolean isMemberOverwriteRequired(MemberShape member) {
+        return (
+            member.getMemberTrait(model, SensitiveTrait.class).isPresent() || isIterationRequired(member)
+        );
+    }
+
+    /**
+     * Returns sanitized value of the member.
```

Comment:
This function doesn't return a value of the member, it returns the member name to be used in generation. Both doc components should be updated.

Reply:
Updated in [commit](https://github.com/awslabs/smithy-typescript/pull/170/commits/dd7fd26dff66268be2fbe4617c541280c589159f)